### PR TITLE
Transformer

### LINKED ISSUE #186: Where are the transformer modules?
## Description
- I have seen transformer's experiment in paper but have not found it in this project
- How to find it and how to use it
- If not, please let us know

### PR DESCRIPTION
**Pull request by benidis**
- Added the `Transformer` model with the deepar feature processing/input.
- Added model tests.

The `layers` module could be moved under `block` but this is up for discussion (currently only used by the `Transformer`).

Fixes #186 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-206/2 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-206/2/index.html

**Comment by szha** [comment]
Job PR-206/3 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-206/3/index.html

**Comment by szha** [comment]
Job PR-206/4 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-206/4/index.html

**Comment by mbohlkeschneider** [comment]
Cool! Do we already have benchmark results on this?

**Comment by benidis** [comment]
We have (KDD table) but from the previous version (swist) which is identical to this one basically.

**Comment by mbohlkeschneider** [comment]
I would suggest to run the benchmark at least once to confirm. Many things changed and might impact performance.

**Comment by geoalgo** [comment]
I agree with Michael, we can proceed in two steps though as a working implementation may already be useful.

**Comment by szha** [comment]
Job PR-206/5 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-206/5/index.html

**Comment by benidis** [comment]
I added a row to the evaluation table for the `TransformerEstimator`. The results are not optimal since I have used default parameters without any hpo but nevertheless they are in the ballpark.

**Comment by szha** [comment]
Job PR-206/6 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-206/6/index.html

**Comment by szha** [comment]
Job PR-206/7 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-206/7/index.html

File: test/model/test_models_synthetic.py [hunk 1]
```diff
@@ -108,4 +131,6 @@ def test_accuracy(Estimator, hyperparameters, accuracy):
         train_dataset=train_ds, test_dataset=test_ds, forecaster=estimator
     )
 
+    print(agg_metrics["ND"])
+    print(accuracy)
```
Comment (lostella) [inline test/model/test_models_synthetic.py]: these prints can be removed?

File: evaluations/README.md [hunk 2]
```diff
@@ -38,6 +44,7 @@ RForecastPredictor_arima |  | 0.011 | 641.476 | 2285.035 |  | 1436.552 | 644.820
 RForecastPredictor_ets | 3195.747 | 0.012 | 602.283 | 2158.406 | 1413.275 | 1374.529 | 659.644 | 2066.347 | 65.986 | 0.039
 SeasonalNaivePredictor | 1139.925 | 0.013 | 705.425 | 1901.146 | 1628.794 | 1577.303 | 673.443 | 2016.458 | 62.518 | 0.037
 SimpleFeedForwardEstimator | 1285.875 | 0.014 | 677.479 | 2323.024 | 1420.506 | 1453.103 | 672.740 | 1982.234 | 37.251 | 0.034
+TransformerEstimator | 2059.355 | 0.014 | 664.324 | 1575.836 | 1512.068 | 1494.884 | 848.616 | 2010.936 | 35.152 | 0.026
```
Comment (lostella) [inline evaluations/README.md]: Can you add the corresponding JSONs for this?

**Comment by szha** [comment]
Job PR-206/8 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-206/8/index.html

**Comment by szha** [comment]
Job PR-206/9 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-206/9/index.html

**Pull request merged**